ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableTagSpec.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.parse;
+
+import com.google.common.base.MoreObjects;
+
+public class AlterTableTagSpec <T>{
```

Comment:
why do you need a generic type here? i don't see it used anywhere (same for AlterTableBranchSpec)

Reply:
https://github.com/apache/hive/pull/4372/files#diff-ca555c7ed1f86abc1a58feeb22bf661a84c657a3791944c872a94d68c8309e8eR32  It is used here: Line-32.
I want to use the generic type to delegate multiple alter tag operation, e.g. `CreateTagSpec` and the further work
 `DropTagSpec` & `ReplaceTagSpec`.

Reply:
ok, makes sense, but I still do not see where type information is being used
https://github.com/apache/hive/pull/4372/files#diff-eac7a1ce623e75ae7789cba72422c240ec9d4f877db53280251a6ad9816e8bf5R110 uses a diamond operator, but declared var is missing the generic type.

Reply:
Sorry for my less rigorous coding. Fixed.